File: Backend-API/models/event.py
from datetime import datetime
import uuid

class Event:
    @staticmethod
    def create(user_id, event_type, event_data):
        return {
            '_id': str(uuid.uuid4()),
            'user_id': user_id,
            'event_type': event_type,
            'event_data': event_data,
            'timestamp': datetime.now()
        }

File: RewardXLibrary/src/main/java/com/example/rewardxlibrary/models/Event.kt
package com.example.rewardxlibrary.models

import com.google.gson.annotations.SerializedName

data class Event(
    @SerializedName("_id")
    val id: String? = null,

    @SerializedName("user_id")
    val userId: String,

    @SerializedName("event_type")
    val eventType: String,

    @SerializedName("event_data")
    val eventData: Map<String, Any>,

    @SerializedName("timestamp")
    val timestamp: String? = null
)